api_app/db/repositories/resources.py

```diff
@@ -125,6 +131,27 @@ def patch_resource(self, resource: Resource, resource_patch: ResourcePatch, reso
         self.update_item_with_etag(resource, etag)
         return resource, resource_template
 
+    def validate_template_version_patch(self, resource: Resource, resource_patch: ResourcePatch, resource_template_repo: ResourceTemplateRepository, resource_template: ResourceTemplate, force_version_update: bool = False):
+        parent_resource_id = None
+        if resource.resourceType == ResourceType.UserResource:
+            parent_resource_id = resource.parentWorkspaceServiceId
+
+        # validate Major upgrade
+        desired_version = semver.VersionInfo.parse(resource_patch.templateVersion)
```

Comment:
I was wondering...do we really need a new package installed just to parse and compare the version numbers?
It does a lot more stuff than we need.

However...leave it. It's fine.

Reply:
Yeah I know what you mean, I agree with you that I can write a simple code by myself,
and remove dependency of other libraries, with all the consequences of it (license, security, maintenance etc.),
but anyway @tamirkamara just noticed that we already use another library `semantic-version`,
so I've replaced my usage with that library and remove the dependency of semver.